pipeline_tools/shared/submission/create_analysis_protocol.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+import argparse
+import json
+import os
+import uuid
+from pipeline_tools.shared.schema_utils import SCHEMAS
+from pipeline_tools.shared.submission import format_map
+
+
+class AnalysisProtocol():
+    """AnalysisProtocol class implements the creation of a json analysis protocol for Optimus and SS2 pipeline outputs
+
+
+    HCA system consumes json files that describe the optimus analysis outputs of Optimus and SS2 pipelines.
+
+
+    The json files have the following form:
+
+
+        {
+        "computational_method": "https://dockstore.org/workflows/github.com/broadinstitute/warp/Optimus:Optimus_v4.2.3",
+        "describedBy": "https://schema.humancellatlas.org/type/protocol/analysis/9.1.0/analysis_protocol",
+        "protocol_core": {
+            "protocol_id": "optimus_v4.2.3"
+        },
+        "provenance": {
+            "document_id": "f2cdb4e5-b439-5cdf-ac41-161ff39d5790",
+            "submission_date": "2021-05-24T12:00:00.000000Z",
+            "update_date": "2021-05-24T12:00:00.000000Z"
+        },
+        "schema_type": "protocol",
+        "type": {
+            "text": "analysis_protocol"
+        }
+        }
+
+
+    See https://schema.humancellatlas.org/type/protocol/analysis/9.1.0/analysis_protocol for full spec
+    """
+
+    # All analysis files will share these attributes
+    describedBy = SCHEMAS["ANALYSIS_PROTOCOL"]["describedBy"]
+    schema_type = SCHEMAS["ANALYSIS_PROTOCOL"]["schema_type"]
+    schema_version = SCHEMAS["ANALYSIS_PROTOCOL"]["schema_version"]
+    type = {
+        "text": "analysis_protocol"
+    }
+
+    def __init__(
+        self,
+        input_uuid,
+        method,
+        pipeline_version,
+        pipeline_type,
+            workspace_version):
+
+        self.input_uuid = input_uuid
+        self.computational_method = method
+        self.protocol_core = {
+            "protocol_id": pipeline_version
+        }
+        self.pipeline_type = pipeline_type
+        self.provenance = {
+            "submission_date": workspace_version,
+            "update_date": workspace_version
+        }
+
+        string_to_hash = json.dumps(self.get_json())
```

Comment:
So the way they hash this is stupid, basically they are hashing the string representation of the json body....but the way its being done here is creating a circular dependency so we will have to break this up...we can do it in mobbing